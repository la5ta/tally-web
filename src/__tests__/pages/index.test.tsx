import indexPageQuery from '@queries/indexPage';
import React from 'react';
import { QueryRenderer } from 'react-relay';
import ReactTestRenderer from 'react-test-renderer';
import {
  createMockEnvironment,
  MockPayloadGenerator,
  RelayMockEnvironment
} from 'relay-test-utils';
import Index, { getStaticProps } from '../../pages/index';

describe('Index', () => {
  let testComponent: ReactTestRenderer.ReactTestRenderer;
  let environment: RelayMockEnvironment;

  beforeEach(() => {
    environment = createMockEnvironment();

    const renderQuery = (renderProps) => {
      const users = renderProps.props && renderProps.props.users || [];

      return <Index users={users} />;
    };

    const IndexRenderer = () => (
      <QueryRenderer
        environment={environment}
        query={indexPageQuery}
        variables={{}}
        render={renderQuery}
      />
    );

    testComponent = ReactTestRenderer.create(<IndexRenderer />);
    environment.mock.resolveMostRecentOperation((operation) => MockPayloadGenerator.generate(operation));
  });

  it('should render the index with data', () => {
    expect(testComponent.toJSON()).toMatchSnapshot();
  });

  it('should getStaticProps return an Object', async () => {

    const { Response } = jest.requireActual('node-fetch');
    const ResponseInit = {
      status: 200,
      statusText: 'fail',
      // headers: headers
    };
    const downloadDocumentData = { data: { users: [] } };
    const response: Response = new Response(JSON.stringify(downloadDocumentData), ResponseInit);
    global.fetch = jest.fn(() => Promise.resolve(response));

    const result = await getStaticProps();
    expect(result).toBeInstanceOf(Object);
  });
});