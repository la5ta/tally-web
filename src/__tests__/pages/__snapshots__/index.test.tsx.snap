// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render the index with data 1`] = `
<section
  className="ant-layout ant-layout-has-sider"
  style={
    Object {
      "minHeight": "100vh",
      "overflow": "hidden",
    }
  }
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark show-mobile"
    style={
      Object {
        "flex": "0 0 300px",
        "maxWidth": "300px",
        "minWidth": "300px",
        "width": "300px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <ul
        className="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
        onMouseEnter={[Function]}
        onTransitionEnd={[Function]}
        role="menu"
        style={Object {}}
      >
        <li
          className="ant-menu-submenu ant-menu-submenu-inline"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
        >
          <div
            aria-expanded={false}
            aria-haspopup="true"
            className="ant-menu-submenu-title"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="button"
            style={
              Object {
                "paddingLeft": 24,
              }
            }
          >
            <span
              aria-label="user"
              className="anticon anticon-user"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="user"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                />
              </svg>
            </span>
            <span>
              subnav 1
            </span>
            <i
              className="ant-menu-submenu-arrow"
            />
          </div>
          <div />
        </li>
      </ul>
    </div>
  </aside>
  <main
    className="ant-layout-content min-width-100 blocked"
    onClick={[Function]}
  >
    <header
      className="ant-layout-header"
    >
      <div
        aria-hidden="true"
        className="sider-trigger show-mobile"
        id="button"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          aria-label="menu-unfold"
          className="anticon anticon-menu-unfold"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="menu-unfold"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 000-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0014.4 7z"
            />
          </svg>
        </span>
      </div>
      <div
        className="logo"
        dangerouslySetInnerHTML={
          Object {
            "__html": Object {
              "default": [Function],
            },
          }
        }
      />
    </header>
    <section
      className="ant-layout"
    >
      <section
        className="ant-layout"
      >
        <main
          className="ant-layout-content"
        >
          <div
            className="ant-breadcrumb"
          >
            <span>
              <span
                className="ant-breadcrumb-link"
              >
                Home
              </span>
              <span
                className="ant-breadcrumb-separator"
              >
                /
              </span>
            </span>
            <span>
              <span
                className="ant-breadcrumb-link"
              >
                List
              </span>
              <span
                className="ant-breadcrumb-separator"
              >
                /
              </span>
            </span>
            <span>
              <span
                className="ant-breadcrumb-link"
              >
                App
              </span>
              <span
                className="ant-breadcrumb-separator"
              >
                /
              </span>
            </span>
          </div>
          <div
            className="site-layout-content"
          >
            <div
              className="content"
            >
              <div
                className="ant-row"
                style={Object {}}
              >
                <div
                  className="ant-col ant-col-24"
                  style={Object {}}
                >
                  <h1>
                    Hello there!! 
                    <span
                      aria-label="hand wave"
                      role="img"
                    >
                      ðŸ‘‹
                    </span>
                  </h1>
                  <h3>
                    Users:
                  </h3>
                  <p />
                  <ol>
                    <li>
                      &lt;mock-value-for-field-"first_name"&gt;
                    </li>
                  </ol>
                  <div
                    className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left"
                    role="separator"
                  >
                    <span
                      className="ant-divider-inner-text"
                    >
                      Sin Etiqueta Form
                    </span>
                  </div>
                  <div
                    className="input"
                  >
                    <div
                      className="label"
                    >
                      Usuario
                    </div>
                    <div
                      className="component"
                    >
                      <input
                        className="ant-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="josegale@mitienda.com"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="input"
                  >
                    <div
                      className="label"
                    >
                      ContraseÃ±a
                    </div>
                    <div
                      className="component"
                    >
                      <span
                        className="ant-input-affix-wrapper ant-input-password"
                        onMouseUp={[Function]}
                      >
                        <input
                          action="click"
                          className="ant-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="Contrasenha12345$"
                          style={null}
                          type="password"
                          value=""
                        />
                        <span
                          className="ant-input-suffix"
                        >
                          <span
                            aria-label="eye-invisible"
                            className="anticon anticon-eye-invisible ant-input-password-icon"
                            onClick={[Function]}
                            onMouseDown={[Function]}
                            onMouseUp={[Function]}
                            role="img"
                            tabIndex={-1}
                          >
                            <svg
                              aria-hidden="true"
                              className=""
                              data-icon="eye-invisible"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                              />
                              <path
                                d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                  <div
                    className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left"
                    role="separator"
                  >
                    <span
                      className="ant-divider-inner-text"
                    >
                      Con Etiqueta Form
                    </span>
                  </div>
                  <form
                    className="ant-form ant-form-horizontal"
                    id="basic"
                    onSubmit={[Function]}
                  >
                    <div
                      className="ant-row ant-form-item"
                      style={Object {}}
                    >
                      <div
                        className="ant-col ant-col-8 ant-form-item-label"
                        style={Object {}}
                      >
                        <label
                          className="ant-form-item-required"
                          htmlFor="basic_username"
                          title="Username"
                        >
                          Username
                        </label>
                      </div>
                      <div
                        className="ant-col ant-col-16 ant-form-item-control"
                        style={Object {}}
                      >
                        <div
                          className="ant-form-item-control-input"
                        >
                          <div
                            className="ant-form-item-control-input-content"
                          >
                            <input
                              className="ant-input"
                              id="basic_username"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      className="ant-row ant-form-item"
                      style={Object {}}
                    >
                      <div
                        className="ant-col ant-col-8 ant-form-item-label"
                        style={Object {}}
                      >
                        <label
                          className="ant-form-item-required"
                          htmlFor="basic_password"
                          title="Password"
                        >
                          Password
                        </label>
                      </div>
                      <div
                        className="ant-col ant-col-16 ant-form-item-control"
                        style={Object {}}
                      >
                        <div
                          className="ant-form-item-control-input"
                        >
                          <div
                            className="ant-form-item-control-input-content"
                          >
                            <span
                              className="ant-input-affix-wrapper ant-input-password"
                              onMouseUp={[Function]}
                            >
                              <input
                                action="click"
                                className="ant-input"
                                id="basic_password"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                style={null}
                                type="password"
                                value=""
                              />
                              <span
                                className="ant-input-suffix"
                              >
                                <span
                                  aria-label="eye-invisible"
                                  className="anticon anticon-eye-invisible ant-input-password-icon"
                                  onClick={[Function]}
                                  onMouseDown={[Function]}
                                  onMouseUp={[Function]}
                                  role="img"
                                  tabIndex={-1}
                                >
                                  <svg
                                    aria-hidden="true"
                                    className=""
                                    data-icon="eye-invisible"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                    />
                                    <path
                                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </main>
      </section>
    </section>
    <footer
      className="ant-layout-footer"
    >
      <div
        className="footer"
      >
        <div
          className="column"
        >
          <h2>
            Resources
          </h2>
          <div
            className="item"
          >
            <a
              href="https://pro.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Pro
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://charts.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Charts
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://mobile.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Mobile
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://ng.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              NG-ZORRO
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Ant Design of Angular
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://ng.mobile.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              NG-ZORRO-MOBILE
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://vue.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Vue
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://landing.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Landing
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Landing Templates
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://scaffold.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Scaffolds
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Scaffold Market
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://umijs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Umi
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              React Application Framework
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://d.umijs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Dumi
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Component doc generator
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://remaxjs.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Remax
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Mini Program Framework
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://github.com/alibaba/hooks"
              rel="noopener noreferrer"
              target="_blank"
            >
              ahooks
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              React Hooks Library
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://motion.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Motion
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Motion Solution
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="/docs/resources#Design-Resources"
            >
              Design Resources
            </a>
          </div>
        </div>
        <div
          className="column"
        >
          <h2>
            Resources
          </h2>
          <div
            className="item"
          >
            <a
              href="https://pro.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Pro
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://charts.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Charts
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://mobile.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Mobile
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://ng.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              NG-ZORRO
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Ant Design of Angular
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://ng.mobile.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              NG-ZORRO-MOBILE
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://vue.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Vue
            </a>
          </div>
          <div
            className="item"
          >
            <a
              href="https://landing.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Design Landing
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Landing Templates
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://scaffold.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Scaffolds
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Scaffold Market
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://umijs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Umi
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              React Application Framework
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://d.umijs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Dumi
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Component doc generator
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://remaxjs.org/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Remax
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Mini Program Framework
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://github.com/alibaba/hooks"
              rel="noopener noreferrer"
              target="_blank"
            >
              ahooks
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              React Hooks Library
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="https://motion.ant.design"
              rel="noopener noreferrer"
              target="_blank"
            >
              Ant Motion
            </a>
            <span
              className="separator"
            >
              -
            </span>
            <span
              className="description"
            >
              Motion Solution
            </span>
          </div>
          <div
            className="item"
          >
            <a
              href="/docs/resources#Design-Resources"
            >
              Design Resources
            </a>
          </div>
        </div>
      </div>
    </footer>
  </main>
</section>
`;
